tutorials-v5/time-evolution/013_nonmarkovian_monte_carlo.md

```diff
@@ -0,0 +1,563 @@
+
+The plot shows that the number of trajectories contributing to the average declines rapidly, due to two effects. First, jumps with the jump operators $L_1$ or $L_2$ bring the system into the ground state. A trajectory that enters the ground state at any time cannot leave it again. Second, the solver adds a third jump channel with zero jump rate. A jump in this channel sets the martingale of the corresponding trajectory to zero for the remainder of the time. In the plot below, we visualize this effect by making the trajectories stop before the end of the time interval.
+
+If $\langle H \rangle$ went to zero, a declining number of contributing trajectories would not be an issue. However, due to the negative rates, the ensemble average does not converge to the steady state $|0\rangle\langle 0|$. Therefore, the weights of the remaining contributing trajectories must grow rapidly to counteract the non-contributing trajectories, leading to a rapidly growing error.
+
+Note that the trajectories are plotted with a small offset since otherwise, they would overlap.
+
+```python
+# Adapted from matplotlib tutorial,
+# https://matplotlib.org/stable/gallery/lines_bars_and_markers/multicolored_line.html
+
+# --- Settings ---
+# Lines are plotted w/ offset, spread is difference between max and min offset
+spread = 0.015
+N = 150  # Number of trajectories
+# ----------------
+
+all_traces = [np.abs(tr) for traj in MCSol2.trajectories[0:N]
+              for tr in traj.trace if np.abs(tr) > 0]
+min_martingale = min(all_traces)
+max_martingale = max(all_traces)
+
+fig = plt.figure()
+ax = fig.add_subplot(1, 1, 1)
+norm = LogNorm(min_martingale, max_martingale)
+
+for i, traj in enumerate(MCSol2.trajectories[0:N]):
+    offset = -spread / 2 + spread * i / N
+
+    traj_defined = np.abs(traj.trace) > 0
+    traj_times = times2[traj_defined]
+    traj_trace = np.array(traj.trace)[traj_defined]
+    traj_ex0 = traj.expect[0][traj_defined] / traj_trace
+    traj_ex1 = traj.expect[1][traj_defined] / traj_trace
+    traj_ex2 = traj.expect[2][traj_defined] / traj_trace
+    traj_exH = omeg1 * traj_ex1 + omeg2 * traj_ex2
+
+    points = np.array([traj_times, traj_exH + offset]).T.reshape(-1, 1, 2)
+    segments = np.concatenate([points[:-1], points[1:]], axis=1)
+    colors = np.abs(traj_trace[:-1])
+
+    lc = LineCollection(segments, cmap='PuBu', norm=norm)
+    lc.set_array(colors)
+    line = ax.add_collection(lc)
+
```

Comment:
Maybe also nice to show the expectation value from the monte-carlo solver:

```
ax.plot(times2, omeg1 * MCSol2.average_expect[1] + omeg2 * MCSol2.average_expect[2],
        color='blue', label=r"$\langle H \rangle$ (MC)")
```

Reply:
Added both suggestions :)